plugins/woocommerce/tests/php/src/Blocks/Domain/Services/AddressProviderServiceTest.php

```diff
@@ -0,0 +1,435 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Tests\Blocks\Domain\Services;
+
+use Mockery\Adapter\Phpunit\MockeryTestCase;
+use Automattic\WooCommerce\Blocks\Domain\Services\AddressProviderService;
+use Automattic\WooCommerce\Blocks\Package;
+use WC_Address_Provider;
+
+/**
+ * Tests for Address Provider Service functionality
+ */
+class AddressProviderServiceTest extends MockeryTestCase {
```

Comment:
If we do ensure IDs are unique we can write an additional test case for this.

Reply:
Done in 2648eab807f887a13f718c495357f4730de5eb85